doc/new-provider.md

```diff
@@ -0,0 +1,87 @@
+# How to add a new provider
+
+Adding a new Teller provider is very easy, but you still need to know where to start. we summarize the steps very shortly to make your life easier  
+
+
+## Provider implementation
+
+1. Copy the file [example.go](pkg/providers/example.go) from path [pkg/providers/example.go](../pkg/providers/example.go) and replace all the `example` places with your provider name. make sure the uppercase.
+2. Go to [pkg/providers.go](../pkg/providers.go) and add your provider name to `ProviderHumanToMachine` hashMap. 
+```go
+func (p *BuiltinProviders) ProviderHumanToMachine() map[string]string {
+    return map[string]string{
+        "Heroku":  "heroku",
+        ...
+        "Example": "example"
+    }
+}
+```
+3. Add your provider to switch case providers in [pkg/providers.go](../pkg/providers.go) file.
+```go
+func (p *BuiltinProviders) GetProvider(name string) (core.Provider, error) {
+    logger := logging.GetRoot().WithField("provider_name", name)
+    switch name {
+    case "heroku":
+        return providers.NewHeroku(logger)
+    case "example":
+        return providers.NewExample(logger)
+    default:
+        return nil, fmt.Errorf("provider '%s' does not exist", name)
+    }
+}
+```
+4. Add provider template configuration in path: [pkg/wizard_template.go](../pkg/wizard_template.go)
+```go
+{{- if index .ProviderKeys "example" }}
+  # Add here some authentication required, like list of environment variables
+  example:
+    env_sync:
+       path: redis/config
+    env:
+      ETC_DSN:
+        path: redis/config/foobar
+
+{{end}}
+```
+
```

Comment:
Consider adding an additional step for adding documentation for the new provider, since all the providers are listed in the main readme file with details and examples.